qiskit_ibm_runtime/visualization/utils.py

```diff
@@ -0,0 +1,855 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=too-many-return-statements
+
+"""
+Utility functions for visualizing qiskit-ibm-runtime's objects.
+"""
+
+from typing import List
+
+
+def get_qubits_coordinates(num_qubits: int) -> List[List[int]]:
```

Comment:
Consider a helper with the following signature:

```python

_Row = namedtuple("_Row", "num_qubits,x_offset,down_links")
"""Tuple specifying a dense horizontal row of the standard heavy hex planar embedding.

Args:
    num_qubits: The number of qubits in the dense row.
    x_offset: The integer x-offset of the row.
    down_links: The positions of dense row, wrt the first qubit, at which to place qubits below.
"""

def _heavy_hex_coords(rows: list[_Row], row_major: bool) -> list[tuple[int, int]]:
    """Generate heavy hex coordinates for the given rows.

    Args:
        rows: The dense rows specifications.
        row_major: Whether qubits should be ordered by their ``(x, y)`` locations (as opposed to ``(y, x)``).

    Returns:
        A list of qubit coordinates, where list position corresponds with qubit index.
    """
```

Under this scheme, something like bangkok (27) can be specified as:
```
_heavy_hex_coords(
    [_Row(0, 0, [3, 7]), _Row(10, 0, [1, 5, 9]), _Row(10, 1, [2, 6])],
    False
)
```
Noting that the first _Row with no qubits is to specify qubits 6 and 17 here: 
![image](https://github.com/user-attachments/assets/07bf5db7-26b5-4ee9-aadc-7f7118beed62)


Likewise, our 127s can be specified as
```
_heavy_hex_coords(
    [
        _Row(14, 0, [4, 8, 12]), 
        a:=_Row(15, 0, [2, 6, 10, 14]), 
        b:=_Row(15, 0, [4, 8, 12]), 
        a,
        b,
        a
        _Row(14, 1, []), 
    ],
    True
)
```

If you find a better scheme, go nuts, but this seems easier than keeping track of huge lists.

Reply:
I thought of creating a nice interface, but I think that should live in qiskit rather than on qiskit-ibm-runtime.

Right now, qiskit uses big lists too, and it defines them inside a function ([here](https://github.com/Qiskit/qiskit/blob/main/qiskit/visualization/gate_map.py#L96)], so I cannot even import them, I am forced to copy paste them.

If we add an interface in this PR, I think we should do it with the idea of porting it to qiskit, and that is a project in and of its own (people may see it and come back with comments, requests to change or add stuff..). 
I think that a more sensible way forward may be the following:
- Use big lists in this PR, same as qiskit
- Open a PR in qiskit to add an interface that allows getting rid of the maps
- Once available, start using the interface in qiskit-ibm-runtime

Any thoughts?

Reply:
>  I think that should live in qiskit rather than on qiskit-ibm-runtime[...]we should do it with the idea of porting it to qiskit[...] Open a PR in qiskit
I'm not so sure about this because qiskit is intended to be hardware agnostic, whereas qiskit-ibm-runtime is specifically focused on IBM hardware.

Reply:
Mmmh okay.. This changes things then

Reply:
As agreed privately, I'll do this in a subsequent PR